pylabrobot/liquid_handling/backends/hamilton/STAR.py

```diff
@@ -7317,6 +7318,257 @@ async def clld_probe_z_height_using_channel(
+    else:
+      channel_idx_plus_one_y_pos = 6
+      # Insight: STAR machines appear to lose connection to a channel below y-position=6 mm
+
+    max_safe_upper_y_pos = channel_idx_minus_one_y_pos - 9
+    max_safe_lower_y_pos = channel_idx_plus_one_y_pos + 9 if channel_idx_plus_one_y_pos != 0 else 6
+
+    # Enable safe start and end positions
+    if start_pos_search:
+      assert max_safe_lower_y_pos <= start_pos_search <= max_safe_upper_y_pos, (
+        f"Start position for y search must be between \n{max_safe_lower_y_pos} and "
+        + f"{max_safe_upper_y_pos} mm, is {end_pos_search} mm. Otherwise channel will crash."
+      )
+      await self.move_channel_y(y=start_pos_search, channel=channel_idx)
+
+    if end_pos_search:
+      assert max_safe_lower_y_pos <= end_pos_search <= max_safe_upper_y_pos, (
+        f"End position for y search must be between \n{max_safe_lower_y_pos} and "
+        + f"{max_safe_upper_y_pos} mm, is {end_pos_search} mm. Otherwise channel will crash."
+      )
+
+    current_channel_y_pos = await self.request_y_pos_channel_n(channel_idx)
+    current_channel_z_pos = await self.request_z_pos_channel_n(channel_idx)
+    if tip_len is None:
+      tip_len = await self.request_tip_len_on_channel(channel_idx=channel_idx)
+      await self.move_channel_z(z=current_channel_z_pos, channel=channel_idx)
+
+    tip_len_to_tip_bottom_diameter_dict = {
+      29.9: 0.8,  # 10 ul tip
+      50.4: 0.7,  # 50 ul tip
+      59.9: 1.2,  # teaching needle - BUT not 300 ul tip!
+      95.1: 1.2,  # 1000 ul tip
+    }
+    tip_bottom_diameter = tip_len_to_tip_bottom_diameter_dict[tip_len]
+
+    # Set safe y-search end position based on the probing direction
+    if probing_direction == "forward":
+      max_y_search_pos = end_pos_search or max_safe_upper_y_pos
+      if max_y_search_pos < current_channel_y_pos:
+        raise ValueError(
+          f"Channel {channel_idx} cannot move forwards: "
+          f"End position = {max_y_search_pos} < current position = {current_channel_y_pos}"
+          f"\nDid you mean to move backwards?"
+        )
+    else:  # probing_direction == "backwards"
+      max_y_search_pos = end_pos_search or max_safe_lower_y_pos
+      if max_y_search_pos > current_channel_y_pos:
+        raise ValueError(
+          f"Channel {channel_idx} cannot move backwards: "
+          f"End position = {max_y_search_pos} > current position = {current_channel_y_pos}"
+          f"\nDid you mean to move forwards?"
+        )
+
+    # Convert mm to increments
+    max_y_search_pos_increments = STAR.mm_to_y_drive_increment(max_y_search_pos)
+    channel_speed_increments = STAR.mm_to_y_drive_increment(channel_speed)
+
+    # Machine-compatability check of calculated parameters
+    assert 0 <= max_y_search_pos_increments <= 13_714, (
+      "Maximum y search position must be between \n0 and"
+      + f"{STAR.y_drive_increment_to_mm(13_714)+9} mm, is {max_y_search_pos_increments} mm"
+    )
```

Comment:
we check that the y_pos in increments is lower than 13_714, but the error message says `STAR.y_drive_increment_to_mm(13_714)+9`. It would be good to compute this value once, store in a variable, and use throughout the rest of the method. Currently, the magic number 13_714 (or +9?) appears multiple times